gamemode/modules/f1menu/cl_searchbox.lua

```diff
@@ -6,15 +6,18 @@ function PANEL:Init()
     self.BaseClass.Init(self)
     self.F1Down = true
     self:SetFont("DarkRPHUD2")
-    self:SetTextColor(Color(255,255,255,255))
-    self:SetCursorColor(Color(255,255,255,255))
 
     self.lblSearch = vgui.Create("DLabel", self)
     self.lblSearch:SetFont("DarkRPHUD2")
-    self.lblSearch:SetColor(Color(200, 200, 200, 200))
     self.lblSearch:SetText(DarkRP.getPhrase("f1Search"))
     self.lblSearch:SizeToContents()
     self.lblSearch:SetPos(5)
+
+    if GAMEMODE.Config.DarkRPSkin == "DarkRP" then
+        self:SetTextColor(Color(255,255,255,255))
+        self:SetCursorColor(Color(255,255,255,255))
+        self.lblSearch:SetColor(Color(200, 200, 200, 200))
+    end
```

Comment:
> Yeah, it shouldn't do. I believe it's to do with SetSkin being used on panels that change the skin to something different than what ForceDermaSkin is set to use.

Ah, `SetSkin` will have not yet applied when `PANEL:Init` is called while `ForceDermaSkin` will have. It should be fine later on though so it should have no visual effect unless the control is broken and doesn't fetch the SKIN fields after `PANEL:Init`.

But you did remind me that it was DTextEntry that was broken. See https://github.com/garrynewman/garrysmod/pull/1141. ETA on fix is probably 2018-19 I'm afraid though.

Luckily, there is a very simply workaround which would be to apply the `SKIN:PaintTextEntry` changes into the DarkRP skin. It is forwards compatible, requires a minuscule amount of effort, and will have the exact same effect as if it was incorporated in the default GMod skin. I wouldn't even call it "hacky" either. There are much hackier workarounds in DarkRP currently. Of course, it all comes down to what @FPtje prefers though.

---

To clarify the other points:

> whilst providing a little more flexibility for others wishing to use their own skin

Changes to the DarkRP skin does not affect this in any way. Baseclasses don't exist in the SKIN world.

> The change made here shouldn't cause conflictions with anything at least

Neither solution should cause conflicts.

> although a global variable could be defined which holds the name instead (which the skin itself uses for the name), but I doubt it's worth it

Definitely not.

> the whole "staying local" idea

Addressed in my previous comment. It's not a bad mindset though in some cases. I just disagree in it applying here.